build/build.go

```diff
@@ -1235,6 +1238,33 @@ func LoadInputs(ctx context.Context, d driver.Driver, inp Inputs, pw progress.Wr
 			target.FrontendAttrs["context:"+k] = v.Path
 			continue
 		}
+
+		// handle OCI layout
+		if strings.HasPrefix(v.Path, "oci-layout://") {
+			pathAlone := strings.TrimPrefix(v.Path, "oci-layout://")
+			// need to add "dummy.io", because reference.Parse*() funcs require we do not start with a '/'
+			named, err := reference.ParseNormalizedNamed(fmt.Sprintf("dummy.io%s", pathAlone))
+			if err != nil {
+				return nil, errors.Wrapf(err, "invalid oci-layout context %s, must be oci-layout:///path/to/layout@sha256:hash", v.Path)
+			}
+			// check that we had a digest
+			digested, ok := named.(reference.Digested)
```

Comment:
I wasn't sure where this goes. In the client test in buildkit [here](https://github.com/moby/buildkit/blob/a6a114a1a476c99c2501aa34811159b849df4005/client/client_test.go#L1669-L1689), we pulled the digest (which I can get from `digested.Digest()`) and put it in `llb.OCILayout(storeID, digest)`, which is used in the `client.Solve(ctx, def, opts)` in `def`.

Where do we put the digest here in buildx so it knows that the build-context `foo` refers not only to the OCI store, but to a specific digest?